src/DotNetWorker.ApplicationInsights/FunctionsTelemetryModule.cs

```diff
@@ -40,10 +66,64 @@ public void Initialize(TelemetryConfiguration configuration)
             ActivitySource.AddActivityListener(_listener);
         }
 
-        public void Dispose()
+        private static void TrackExceptionTelemetryFromActivityEvent(ActivityEvent activityEvent, TelemetryClient telemetryClient)
+        {
+            if (activityEvent.Name == TraceConstants.AttributeExceptionEventName)
+            {
+                string? exceptionType = null;
+                string? exceptionStackTrace = null;
+                string? exceptionMessage = null;
+
+                foreach (var tag in activityEvent.Tags)
+                {
+                    if (tag.Key == TraceConstants.AttributeExceptionType)
+                    {
+                        exceptionType = tag.Value?.ToString();
+                        continue;
+                    }
+                    if (tag.Key == TraceConstants.AttributeExceptionMessage)
+                    {
+                        exceptionMessage = tag.Value?.ToString();
+                        continue;
+                    }
+                    if (tag.Key == TraceConstants.AttributeExceptionStacktrace)
+                    {
+                        exceptionStackTrace = tag.Value?.ToString();
+                        continue;
+                    }
+                }
+
+                ExceptionDetailsInfo edi = new(1, -1, exceptionType, exceptionMessage, exceptionStackTrace != null,
+                    exceptionStackTrace, Enumerable.Empty<Microsoft.ApplicationInsights.DataContracts.StackFrame>());
+
+                ExceptionTelemetry et = new(new[] { edi }, SeverityLevel.Error, null, new Dictionary<string, string>() { }, new Dictionary<string, double>() { });
+
+                telemetryClient.TrackException(et);
+            }
+        }
+
+        public async ValueTask DisposeAsync()
         {
-            _telemetryClient?.Flush();
             _listener?.Dispose();
+
+            if (_telemetryClient is not null)
+            {
+                CancellationTokenSource cts = new(millisecondsDelay: 5000);
```

Comment:
small nit:

```suggestion
                using CancellationTokenSource cts = new(millisecondsDelay: 5000);
```

and remove the `finally`?